components/account/AccountFollowButton.vue

```diff
@@ -65,6 +68,52 @@ const buttonStyle = $computed(() => {
   // If not following, use a button style
   return 'text-inverted bg-primary border-primary'
 })
+
+const dataGlean = $computed(() => {
+  if (!gleanContext)
+    return undefined
+
+  let action
+  if (relationship?.blocking)
+    action = 'follow.unblock'
+  else if (relationship?.muting)
+    action = 'follow.unmute'
+  else if (relationship ? relationship.following : context === 'following')
+    action = 'follow.unfollow'
+  else if (relationship?.requested)
+    action = 'follow.withdraw-follow-request'
+  else if ((relationship ? relationship.followedBy : context === 'followedBy') && account.locked)
+    action = 'follow.follow-request'
+  else
+    action = 'follow.follow'
+
+  return `${gleanContext}.${action}`
+})
+
+function handleClick() {
```

Comment:
I did some clever ordering with the contents of this function. I'm open to some non-clever alternatives

Reply:
The reordering of contents in this function might be causing the bug where the `unfollow` modal shows up when trying to `unmute` someone

Reply:
Ooh good catch. I was unaware of this "mixed" use case. That counts as a 7th state for this horrible button component :'(

Reply:
Let's make sure that whatever changes we make, the end result matches what is on production. We don't want to introduce any usability differences in this PR, only analytics events. We can follow up with improvements to this terrible button in a future PR.

Reply:
Pushed up fixing the bug by removing my cleverness and bringing back the original order of events

Reply:
Verified that I'm no longer seeing that bug!